spec/predicates/deployment.md

```diff
@@ -0,0 +1,461 @@
+- The scope match verification succeeds, because:
+  - Root 1: The service account (in the first attesttation) matches the environment the container is to be deployed
+  - Root 2: The cluster ID (in the second attesttation) matches the environment the container is to be deployed.
+
+### Example 6: Implicit scope type
+
+**Trusted (single) root configuration**:
+- A public key (ignored by the example)
+- Authoritative scope types: "binaryauthorization.googleapis.com/policy_uri" and "kubernetes.io/pod/namespace"
+- Required scope types: "binaryauthorization.googleapis.com/policy_uri" with value "projects/foo/platforms/gke/policies/bar".
+
+**Deployment request**:
+- For a container running in a Kubernetes namespace "prod-namespace"
+- With the following attestation matching the container's digest in the request:
+
+```jsonc
+{
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [{ ... }],
+
+  "predicateType": "https://in-toto.io/attestation/deployment/v1",
+  "predicate": {
+    "creationTime": "...",
+    "scopes": {
+      "kubernetes.io/pod/namespace/v1": "prod-namespace",
+      "binaryauthorization.googleapis.com/policy_uri": "projects/foo/platforms/gke/policies/bar"
+    }
+}
+```
+
+**Verification result**:
+- The attestation authentication verification succeeds, because the trusted root
+  is authoritative for scope types "kubernetes.io/pod/namespace/v1" and "binaryauthorization.googleapis.com/policy_uri".The only required scope type is "binaryauthorization.googleapis.com/policy_uri" and is present in the attestation.
+- The scope match verification succeeds, because the namespace matches the environment and
+the "binaryauthorization.googleapis.com/policy_uri" value matches the configuration value "projects/foo/platforms/gke/policies/bar".
+
+### Example 7: Unrecognized scope
+
+**Trusted (single) root configuration**:
+- A public key (ignored by the example)
+- Authoritative scope types: "cloud.google.com/service_account"
+- Required scope types: "cloud.google.com/service_account"
+
+**Deployment request**:
+- For a container running under service account "sa-name@project.iam.gserviceaccount.com"
+- With the following attestation matching the container's digest in the request:
+
+```jsonc
+{
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [{ ... }],
+
+  "predicateType": "https://in-toto.io/attestation/deployment/v1",
+  "predicate": {
+    "creationTime": "...",
+    "scopes": {
+      "cloud.google.com/service_account/v1": "sa-name@project.iam.gserviceaccount.com",
+      "my.custom-scope.com/some-field/v1": "some-value"
+    }
+}
+```
+
+**Verification result**:
+- The attestation authentication verification fails, because the trusted root
+  is _not_ authoritative for the unrecognized scope type "my.custom-scope.com/some-field/v1".
+- The scope match verification is _not_ performed, because the authentication
+verification failed.
+
+### Example 8: No scope
```

Comment:
no scope is essentially the same as having no `optionally conditions attached` per your comment https://github.com/in-toto/attestation/pull/341/files#r1559618458. Please clarify if I'm mis-understanding.

Reply:
I think the difference in mental model for how this is enabled.

It seems like in your view, each resource (cluster, service account, etc) is configured individually for which authorities ought to be required, and it's normal to have zero, one, or more than one authority. In this model, it makes sense for an authority to sign things with no `scopes` because mere presence of a DA is sufficient to say "good". For example, one DA might say "tested". This case actually devolves to a VSA.

In my view, each resource has the *same* configuration, usually with a single authority. For example, _every_ GCP project under the org example.com might just say "Require a DA from ExampleComAuthority". Because that authority is so broad, it doesn't make sense to omit the `resource` scope. You would almost never have a situation where an image is allowed on every cluster everywhere. (I can imagine it in a few edge cases, but we shouldn't optimize for that.) So that's why I was thinking `resource` would be required.

Reply:
> For example, one DA might say "tested". This case actually devolves to a VSA.

That's correct.

The main motivation is that even with a central authority, different teams will have different levels of onboarding. Certain teams may be binding to a service account, others may find it easier to bind to a cluster ID for UX reasons, others may not even be at the stage where they can enforce additional delpoyment conditions. I think it's useful to have this flexibility rather than taking the view that an org is monolithic. Binauthz solution is an example where they need this flexibility.